src/etherscan/verifyCollectionPartyContract.ts

```diff
@@ -0,0 +1,247 @@
+import axios from "axios";
+import { PartyEvent } from "types";
+import { config } from "../config";
+import { NonReceivableInitializedProxySourceCode } from "./constants";
+import {
+  getEtherscanVerifyContractGuid,
+  setEtherscanVerifyContractGuid,
+} from "./helpers";
+
+/**
+ * This module verifies the NonReceivableInitializedProxy contract for collection parties. The following steps
+ * are taken to verify the source code:
+ *   1. fetch the NonReceivableInitializedProxy contract creation bytecode from the collection party creation tx
+ *      using alchemy trace api
+ *   2. extract the constructor arguments bytecode from the contract creation bytecode
+ *   3. construct params needed to verify the contract source on etherscan
+ *   4. call etherscan verify contract api with params
+ */
+
+type AlchemyTrace = {
```

Comment:
nit but could potentially use https://www.npmjs.com/package/@alch/alchemy-web3 instead of having to define our own types here

Reply:
good to know. i can install this in a follow up